tpu-provisioner/internal/cloud/gke_test.go

```diff
@@ -691,6 +691,38 @@ func TestNodePoolForPod(t *testing.T) {
 				},
 			},
 		},
+		{
+			desc: "confidential disk configured in cluster context",
+			gkeContext: GKEContext{
+				NodeConfidentialStorage: true,
+				NodeDiskType:            "hyperdisk-balanced",
+				NodeBootDiskKMSKey:      "my-kms-key",
```

Comment:
I think KMS keys are formatted like: "projects/KEY_PROJECT_ID/locations/LOCATION/keyRings/RING_NAME/cryptoKeys/KEY_NAME"

I don't think it matters for this test though. Feel free to resolve as fixed.